NO-JIRA: testing revbump at startup

tjungblu opened the pull request:
/hold

dusk125 commented on server/revbump/revbump.go:
```diff
@@ -0,0 +1,65 @@
+
+	latest.main += amount
+	latest.sub = 0
+	k := make([]byte, 17)
```
nit: Could do revBytesLen instead of 17

tjungblu replied:
good catch, I was just copy and pasting it over to make the dependency graph happy

tjungblu replied:
fixed

dusk125 reviewed (commented):
One nit otherwise lgtm

tjungblu committed:
DOWNSTREAM: <carry>: ETCD-696: Add rev bumping to force-new-cluster

force-new-cluster seems to have similar watch cache issues as the
ordinary snapshot restore. This PR introduces the already existing utl logic
as a separate package into the server-side code.

This will only introduce a revbump flag, but under the hood implement
both rev bumping and compaction markers.

Signed-off-by: Thomas Jungblut <tjungblu@redhat.com>
